packages/atomic/src/utils/accessibility-utils.tsx

```diff
@@ -17,7 +18,58 @@ export function AriaLiveRegion(regionName: string) {
 
   return (component: InitializableComponent, setterName: string) => {
     Object.defineProperty(component, setterName, {
-      set: (message) => dispatchMessage(message),
+      set: (message: string) => dispatchMessage(message),
     });
   };
 }
+
+export interface FocusTargetController {
+  setTarget(element: HTMLElement | undefined): void;
+  focusAfterSearch(): void;
+}
+
+export function FocusTarget() {
```

Comment:
The amount of issues and instabilities I had while developing this component is wild. I may be able to do some small code improvements to this decorator, but I had so many issues that I can't keep track of which was the reason for the various fixes/workarounds I implemented.

I think that having multiple use-cases for FocusTarget in this same PR really paid off, since lifecycle/ref functions were called differently across multiple use-cases.

Now, after all those fixes, the decorator seems very reliable and behaves predictably.